core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -650,7 +650,7 @@ public void freeAllScrollContexts() {
 
     private void contextScrollKeepAlive(SearchContext context, long keepAlive) throws IOException {
         if (keepAlive > maxKeepAlive) {
-            throw new QueryPhaseExecutionException(context,
+            throw new IllegalArgumentException(
```

Comment:
do you think it would be possible to have a unit test for this too?

Reply:
It's not easy to add unit test for this. The `SearchServiceTests` is also extend `ESSingleNodeTestCase` for other test cases. And this has been covered by the integration test. I can try to add a unit test if you think it's necessary.